examples/common/src/chariott/streaming.rs

```diff
@@ -1,49 +1,30 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-use async_trait::async_trait;
-use chariott_common::proto::common::{
-    fulfillment::Fulfillment, ReadFulfillment, ReadIntent, SubscribeFulfillment, SubscribeIntent,
+use chariott_common::{
+    ess::SharedEss as InnerEss,
+    proto::common::{fulfillment::Fulfillment, ReadFulfillment, ReadIntent, SubscribeIntent},
 };
 use keyvalue::{InMemoryKeyValueStore, Observer};
-use std::{
-    fmt::Display,
-    hash::Hash,
-    sync::{Arc, RwLock},
-    time::SystemTime,
-};
-use tokio::spawn;
-use tokio_stream::wrappers::ReceiverStream;
-use tonic::{Response, Status};
-
-use crate::chariott::proto::{
-    common::Value as ProtoValue,
-    streaming::{channel_service_server::ChannelService, Event as ProtoEvent, OpenRequest},
-};
-
-const METADATA_KEY: &str = "x-chariott-channel-id";
+use std::sync::RwLock;
+use tonic::Status;
 
-type ClientId = Box<str>;
-type Ess<T> = ess::EventSubSystem<ClientId, EventId, (EventId, T), Result<ProtoEvent, Status>>;
+use crate::chariott::proto::{common::value::Value as ProtoValue, common::Value as ValueMessage};
 
-#[derive(Clone, Hash, Eq, PartialEq)]
-pub struct EventId(Box<str>);
+type EventId = Box<str>;
 
-impl<T: Into<Box<str>>> From<T> for EventId {
-    fn from(value: T) -> Self {
-        Self(value.into())
-    }
-}
+#[derive(Clone)]
+struct Ess<T>(InnerEss<(EventId, T)>);
 
-impl Display for EventId {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.0.fmt(f)
+impl<T: Clone + Send + 'static> Observer<EventId, T> for Ess<T> {
+    fn on_set(&mut self, key: &EventId, value: &T) {
+        self.0.as_ref().publish(key, (key.clone(), value.clone()));
     }
 }
 
-impl<T: Clone> Observer<EventId, T> for Arc<Ess<T>> {
-    fn on_set(&mut self, key: &EventId, value: &T) {
-        self.publish(key, (key.clone(), value.clone()));
+impl<T> AsRef<InnerEss<(EventId, T)>> for Ess<T> {
```

Comment:
Like the [other suggestion](https://github.com/eclipse/chariott/pull/15/files#r1012780174), I think you want to implement `Deref` here too:

```diff
diff --git a/examples/common/src/chariott/streaming.rs b/examples/common/src/chariott/streaming.rs
index 99ae11a..848ce52 100644
--- a/examples/common/src/chariott/streaming.rs
+++ b/examples/common/src/chariott/streaming.rs
@@ -6,7 +6,7 @@ use chariott_common::{
     proto::common::{fulfillment::Fulfillment, ReadFulfillment, ReadIntent, SubscribeIntent},
 };
 use keyvalue::{InMemoryKeyValueStore, Observer};
-use std::sync::RwLock;
+use std::{ops::Deref, sync::RwLock};
 use tonic::Status;
 
 use crate::chariott::proto::{common::value::Value as ProtoValue, common::Value as ValueMessage};
@@ -22,8 +22,10 @@ impl<T: Clone + Send + 'static> Observer<EventId, T> for Ess<T> {
     }
 }
 
-impl<T> AsRef<InnerEss<(EventId, T)>> for Ess<T> {
-    fn as_ref(&self) -> &InnerEss<(EventId, T)> {
+impl<T> Deref for Ess<T> {
+    type Target = InnerEss<(EventId, T)>;
+
+    fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
@@ -82,7 +84,7 @@ where
     T: Into<ProtoValue> + Clone + Send + Sync + 'static,
 {
     fn subscribe(&self, subscribe_intent: SubscribeIntent) -> Result<Fulfillment, Status> {
-        let result = self.ess.as_ref().serve_subscriptions(subscribe_intent, |(_, v)| v.into())?;
+        let result = self.ess.serve_subscriptions(subscribe_intent, |(_, v)| v.into())?;
         Ok(Fulfillment::Subscribe(result))
     } 
```

Reply:
Based on your other comment, I removed the `AsRef` instead with 1ce2e6ef20a5b9e76f41fc82675bcf9fcab5423a as the newtype becomes a private implementation detail.